lib/debug/src/read_debuginfo.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::HashMap;
+use wasmparser::{ModuleReader, SectionCode};
+
+use gimli;
+
+use gimli::{
+    DebugAbbrev, DebugAddr, DebugInfo, DebugLine, DebugLineStr, DebugLoc, DebugLocLists,
+    DebugRanges, DebugRngLists, DebugStr, DebugStrOffsets, DebugTypes, EndianSlice, LittleEndian,
+    LocationLists, RangeLists,
+};
+
+trait Reader: gimli::Reader<Offset = usize, Endian = LittleEndian> {}
+
+impl<'input> Reader for gimli::EndianSlice<'input, LittleEndian> {}
```

Comment:
Why a new `Reader` trait instead of using `gimli::Reader` directly? To hide the associated types?

Reply:
I took it from gimli's examples/dwarfdump.rs -- it is the only syntax I could find atm